src/component/auth_providers/mod.rs

```diff
@@ -94,13 +95,35 @@ fn LoginProvButton<Cb: Fn(ev::MouseEvent) + 'static>(
 ) -> impl IntoView {
     let ctx: LoginProvCtx = expect_context();
 
+    let click_action = create_action(move |()| async move {
+        #[cfg(all(feature = "hydrate", feature = "ga4"))]
+        {
+            use crate::utils::event_streaming::send_event;
+
+            // login_method_selected - analytics
+            send_event(
+                "login_method_selected",
+                &json!({
```

Comment:
this is fine for now but I want strongly typed events eventually

Reply:
yup, will do